### PR TITLE
Fix VerifyWitness

### PR DESCRIPTION
**Pull request by Qiao-Jin**
Fix VerifyWitness

### CODE REVIEW

**Comment by shargon** [comment]
Is not executed in test mode

**Comment by Qiao-Jin** [comment]
> Is not executed in test mode

Here is not test mode

**Comment by shargon** [comment]
> Here is not test mode

When is not in test mode, the gas stop the execution and FAULT, when it's 0.

**Comment by Qiao-Jin** [comment]
> > Here is not test mode
> 
> When is not in test mode, the gas stop the execution and FAULT, when it's 0.

Shouldn't we consider accumulated gas consuming for multiple-witness cases, as in this PR?

File: src/neo/SmartContract/Helper.cs [hunk 1]
```diff
@@ -167,6 +167,8 @@ internal static bool VerifyWitnesses(this IVerifiable verifiable, StoreView snap
                     engine.LoadScript(verifiable.Witnesses[i].InvocationScript, CallFlags.None);
                     if (engine.Execute() == VMState.FAULT) return false;
                     if (!engine.ResultStack.TryPop(out StackItem result) || !result.ToBoolean()) return false;
+                    gas -= engine.GasConsumed;
+                    if (gas < 0) return false;
```
Comment (shargon) [inline src/neo/SmartContract/Helper.cs:171]: this line should never be touched.

**Comment by shargon** [comment]
> Shouldn't we consider accumulated gas consuming for multiple-witness cases, as in this PR?

Now I see what you are trying to do, be more explicit in the public relations description.

File: src/neo/SmartContract/Helper.cs (hunk 1 above)
Comment (Qiao-Jin) [reply src/neo/SmartContract/Helper.cs]: Have removed

**Comment by Qiao-Jin** [comment]
> Could you port this PR to neo2?

Sure:)

**Comment by Qiao-Jin** [comment]
> Could you port this PR to neo2?

It seems that we don't need to port this PR to neo2, as in neo2 there is not an input gas for witness verification:

https://github.com/neo-project/neo/blob/75df7cdaeffc84cd116d907ceda11054be696af3/neo/SmartContract/Helper.cs#L114

**Comment by shargon** [comment]
@Qiao-Jin `out-of-date with the base branch`

**Pull request merged**